cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/AlertaSelfHealingNotifier.java

```diff
@@ -0,0 +1,250 @@
+import com.linkedin.kafka.cruisecontrol.detector.TopicPartitionSizeAnomaly;
+import com.linkedin.kafka.cruisecontrol.detector.TopicReplicationFactorAnomaly;
+import com.linkedin.kafka.cruisecontrol.detector.TopicReplicationFactorAnomaly.TopicReplicationFactorAnomalyEntry;
+import com.linkedin.kafka.cruisecontrol.monitor.sampling.holder.BrokerEntity;
+
+public class AlertaSelfHealingNotifier extends SelfHealingNotifier {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AlertaSelfHealingNotifier.class);
+    public static final String ALERTA_SELF_HEALING_NOTIFIER_API_URL = "alerta.self.healing.notifier.api_url";
+    public static final String ALERTA_SELF_HEALING_NOTIFIER_API_KEY = "alerta.self.healing.notifier.api_key";
+    public static final String ALERTA_SELF_HEALING_NOTIFIER_ENVIRONMENT = "alerta.self.healing.notifier.environment";
+    
+    protected String _alertaApiUrl;
+    protected String _alertaApiKey;
+    protected String _alertaEnvironment;
+
+    public AlertaSelfHealingNotifier() {
+    }
+
+    public AlertaSelfHealingNotifier(Time time) {
+        super(time);
+    }
+
+    @Override
+    public void configure(Map<String, ?> config) {
+        super.configure(config);
+        _alertaApiUrl = (String) config.get(ALERTA_SELF_HEALING_NOTIFIER_API_URL);
+        _alertaApiKey = (String) config.get(ALERTA_SELF_HEALING_NOTIFIER_API_KEY);
+        _alertaEnvironment = (String) config.get(ALERTA_SELF_HEALING_NOTIFIER_ENVIRONMENT);
+    }
+
+    @Override
+    public void alert(Anomaly anomaly, boolean autoFixTriggered, long selfHealingStartTime, AnomalyType anomalyType) {
+        super.alert(anomaly, autoFixTriggered, selfHealingStartTime, anomalyType);
+
+        if (_alertaApiUrl == null) {
+            LOG.warn("Alerta.io API URL is null, can't send Alerta.io self healing notification");
+            return;
+        }
+
+        if (_alertaApiKey == null) {
+            LOG.warn("Alerta.io API key is null, can't send Alerta.io self healing notification");
+            return;
+        }
+
+        String text = String.format("%s detected %s. Self healing %s.%s", anomalyType, anomaly,
+                _selfHealingEnabled.get(anomalyType) ? String.format("start time %s", utcDateFor(selfHealingStartTime))
+                        : "is disabled",
+                autoFixTriggered ? "%nSelf-healing has been triggered." : "");
+
+        String tmpLocalHostname;
+        try {
+          tmpLocalHostname = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e1) {
+          LOG.warn("Unable to get the hostname of the Cruise Control server");
+          tmpLocalHostname = "Cruise Control";
+        }
+        final String localHostname = tmpLocalHostname;
+
+        DateTimeFormatter outFormatter = DateTimeFormatter
+            .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX") // millisecond precision
+            .withZone(ZoneId.of("UTC"));
```

Comment:
Not possible as Alerta needs milliseconds precision.

Reply:
I see -- then can we
1. Add the following function to `CruiseControlUtils`:
```
  public static String utcDateFor(long timeMs, TemporalUnit unit) {
    return Instant.ofEpochMilli(timeMs).truncatedTo(unit).toString();
  }
```
2. Then use the response of this function in this class?

When called with `ChronoUnit.MILLIS`, this will return a response in ms precision, such as `2021-05-05T01:14:46.856Z`.